### PR TITLE
bugfix: fix register branch thread hang on tcc mode

### LINKED ISSUE #1888: thread hang on tcc mode
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅱ. Describe what happened

image
image

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

### Ⅰ. Describe what this PR did

bugfix: fix register branch thread hang on tcc mode

### Ⅱ. Does this pull request fix one issue?
fix #1888 #1862 

### Ⅳ. Describe how to verify it
#1401

### CODE REVIEW

File: saga/seata-saga-rm/src/main/java/io/seata/saga/rm/SagaResourceManager.java [hunk 1]
```diff
@@ -28,115 +31,117 @@
...
+                return BranchStatus.PhaseTwo_RollbackFailed_Retryable;
+            } else if (ExecutionStatus.FA.equals(machineInstance.getStatus())
+                && machineInstance.getCompensationStatus() == null) {
+                return BranchStatus.PhaseOne_Failed;
+            }
 
-		} catch (ForwardInvalidException e) {
-			LOGGER.error("StateMachine forward failed, xid: " + xid, e);
+        } catch (ForwardInvalidException e) {
+            LOGGER.error("StateMachine forward failed, xid: " + xid, e);
```
Comment (jsbxyyx) [inline saga/seata-saga-rm/src/main/java/io/seata/saga/rm/SagaResourceManager.java]: use {}

File: saga/seata-saga-rm/src/main/java/io/seata/saga/rm/SagaResourceManager.java [hunk 2]
```diff
@@ -28,115 +31,117 @@
...
+        try {
+            StateMachineInstance stateMachineInstance = StateMachineEngineHolder.getStateMachineEngine().compensate(xid,
+                null);
+            if (ExecutionStatus.SU.equals(stateMachineInstance.getCompensationStatus())) {
                 return BranchStatus.PhaseTwo_Rollbacked;
             }
-		} catch (EngineExecutionException e) {
-			LOGGER.error("StateMachine compensate failed, xid: " + xid, e);
+        } catch (EngineExecutionException e) {
+            LOGGER.error("StateMachine compensate failed, xid: " + xid, e);
```
Comment (jsbxyyx) [inline saga/seata-saga-rm/src/main/java/io/seata/saga/rm/SagaResourceManager.java]: use {}

File: saga/seata-saga-rm/src/main/java/io/seata/saga/rm/SagaResourceManager.java (hunk 1 above)
Comment (slievrly) [reply saga/seata-saga-rm/src/main/java/io/seata/saga/rm/SagaResourceManager.java]: revert.

File: saga/seata-saga-rm/src/main/java/io/seata/saga/rm/SagaResourceManager.java (hunk 2 above)
Comment (slievrly) [reply saga/seata-saga-rm/src/main/java/io/seata/saga/rm/SagaResourceManager.java]: revert.

**Pull request merged**